conda_env/env.py

```diff
@@ -177,11 +177,13 @@ def parse(self):
         if not self.raw:
             return
 
+        self.clear()
```

Comment:
Before these two changes in parse, if pip "section" was specified multiple times in the environment file, all but the last occurrence were silently ignored. Since there were no warnings, I considered it a bug and changed the behaviour. This change now merges dependencies form all pip "sections" into the "pip" key in the dictionary.

If the old behaviour was intentional, I can revert this change, add a warning if config has multiple pip sections and move the merging logic to `add` or `enable_optional_dependencies` function.